src/iop/finalscale.c

```diff
@@ -59,11 +59,10 @@ void modify_roi_in(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const d
 {
   *roi_in = *roi_out;
 
-  roi_in->x /= roi_out->scale;
-  roi_in->y /= roi_out->scale;
-  // out = in * scale + .5f to more precisely round to user input in export module:
-  roi_in->width  = (roi_out->width  - .5f)/roi_out->scale;
-  roi_in->height = (roi_out->height - .5f)/roi_out->scale;
+  roi_in->x = MAX(0, (int)floorf(roi_out->x / roi_out->scale));
+  roi_in->y = MAX(0, (int)floorf(roi_out->y / roi_out->scale));
+  roi_in->width  = (int)ceilf(roi_out->width  / roi_out->scale);
```

Comment:
those two lines are the ones making the testsuite to fail.

Reply:
This is indeed a tricky one.

Currently we effectively reduce the size of roi_in so the roi_in area will we slower than roi_out, so with a scale of 1 there is some sort of mismatch (this btw leads to cpu fallback as the CL 1:1 copy doesn't work.) 

Currently i don't know to handle that, will come back again on this if i can find a proper solution.

Reply:
Yes a tricky one! If there is no better solution we can certainly recreate the 98 expected output PNG in the integration testsuite... But that's for sure something I'd like to avoid.